plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/KVMHAMonitor.java

```diff
@@ -77,84 +92,148 @@ public NfsStoragePool getStoragePool(String uuid) {
         }
     }
 
-    private class Monitor extends ManagedContextRunnable {
-
-        @Override
-        protected void runInContext() {
-            synchronized (_storagePool) {
-                Set<String> removedPools = new HashSet<String>();
-                for (String uuid : _storagePool.keySet()) {
-                    NfsStoragePool primaryStoragePool = _storagePool.get(uuid);
+    protected class CheckPoolThread extends Thread {
```

Comment:
why not make this a ManagemedContextRunnable as well?

Reply:
ManagedContextRunnable and Thread are both implemetation of Runnable interface.
Thread works in testing. 

ManagedContextRunnable should also be ok. The following might work as well
```
protected class CheckPoolThread implements Runnable {
```